lib/VolSurface/Calibration/Equities.pm

```diff
@@ -0,0 +1,637 @@
+
+    return \@guess;
+}
+
+sub _calculate_skew {
+    my ($self, $tiy, $params) = @_;
+
+    my $sig_small  = $params->{skewshort};
+    my $sig_1y     = $params->{skew1year};
+    my $sig_inf    = $params->{skewlong};
+    my $t_small    = 1.0;
+    my $atm_alpha1 = $params->{skewwingL};
+    my $atm_alpha2 = $params->{skewwingR};
+
+    my $skew =
+        $sig_inf +
+        ($sig_1y - $sig_inf) *
+        ((exp(-$atm_alpha2 * $tiy) - exp(-$atm_alpha1 * $tiy)) / (exp(-$atm_alpha2 * $t_small) - exp(-$atm_alpha1 * $t_small))) +
+        ($sig_small - $sig_inf) *
+        ((exp(-$atm_alpha2 * $t_small - $atm_alpha1 * $tiy) - exp(-$atm_alpha1 * $t_small - $atm_alpha2 * $tiy)) /
+            (exp(-$atm_alpha2 * $t_small) - exp(-$atm_alpha1 * $t_small)));
+
+    return $skew;
+}
+
+sub _calculate_kurtosis {
+    my ($self, $tiy, $params) = @_;
+
+    my $kurt_small = $params->{kurtosisshort};
+    my $kurt_alpha = $params->{kurtosisgrowth};
+    my $kurt_inf   = $params->{kurtosislong};
+
+    return $kurt_small + ($kurt_inf - $kurt_small) * (1 - exp(-$kurt_alpha * $tiy));
+}
+
+sub _calculate_variance {
+    my ($self, $tiy, $params) = @_;
+
+    my $sig_small  = $params->{atmvolshort}**2;
+    my $sig_1y     = $params->{atmvol1year}**2;
+    my $sig_inf    = $params->{atmvolLong}**2;
+    my $t_small    = 1.0;
+    my $atm_alpha1 = $params->{atmWingL};
+    my $atm_alpha2 = $params->{atmWingR};
+
+    my $atm_vol =
+        $sig_inf +
+        ($sig_1y - $sig_inf) *
+        ((exp(-$atm_alpha2 * $tiy) - exp(-$atm_alpha1 * $tiy)) / (exp(-$atm_alpha2 * $t_small) - exp(-$atm_alpha1 * $t_small))) +
+        ($sig_small - $sig_inf) *
+        ((exp(-$atm_alpha2 * $t_small - $atm_alpha1 * $tiy) - exp(-$atm_alpha1 * $t_small - $atm_alpha2 * $tiy)) /
+            (exp(-$atm_alpha2 * $t_small) - exp(-$atm_alpha1 * $t_small)));
+
+    return $atm_vol;
+}
+
+# Algorithm change - now based on centroid calculations
+# A function that optimizes a set of parameters against a function.
+# This optimization method is based on Amoeba optimization
+
+sub _optimize {
+    my ($self, $params) = @_;
+
+    my $intol         = $self->_tolerance_level;
+    my $num_of_var    = scalar(@$params);
+    my $num_of_points = $num_of_var;
+
+    my $highest_value;
+
+    my $lambda1 = ((($num_of_points + 1)**0.5) - 1 + $num_of_points) / ($num_of_points * (2**0.5)) * $self->_scale;
+    my $lambda2 = ((($num_of_points + 1)**0.5) - 1) / ($num_of_points * (2**0.5)) * $self->_scale;
+
+    my $i;
+    my $j;
+
+    #initialize the simplex with the initial guess
+    my @simplex = ($params);
+
+    my $step = 0;
+    for ($i = 0; $i < $num_of_var; $i++) {
+        $step = 0.1 * abs($params->[$i])
+            if (0.1 * abs($params->[$i]) > $step);
+    }
+
+    for (my $i = 1; $i <= $num_of_points; $i++) {
+        my @new = map { ($i - 1 == $_) ? $simplex[0][$_] + $step : 0 + $simplex[0][$_] } (0 .. $num_of_var - 1);
+        push @simplex, \@new;
+    }
```

Comment:
Is there any reason why this loop runs from `$i=1` instead of `0`?

```perl
for (my $i=0; $i<$num_of_points; $i++) {
    push @simplex, [map { $simplex[0][$_] + ($i==$_ ? $step : 0) } (0 .. $num_of_var-1)];
}
```